analyzers/src/SonarAnalyzer.SourceGenerators/RuleCatalogGenerator.cs

```diff
@@ -60,7 +60,7 @@ private static IEnumerable<string[]> RuleDescriptorArguments(string rspecDirecto
             foreach (var jsonPath in Directory.GetFiles(rspecDirectory, "*.json").Where(x => Path.GetFileName(x) != SonarWayFileName))
             {
                 var json = JObject.Parse(File.ReadAllText(jsonPath));
-                var id = Path.GetFileName(jsonPath).Split('_').First();
+                var id = Path.GetFileName(jsonPath).Split('.').First();
```

Comment:
Split doesn't make sense anymore, use `Path.GetFileNameWithoutExtension` instead

Reply:
I took a look out of curiosity because I rewrote this part from a giant mess in the past. 

The rest of the changes looks good, it's nice to see this being simplified.